tests/integration/questionnaire/test_questionnaire_previous_link.py

```diff
@@ -0,0 +1,75 @@
+from tests.integration.create_token import create_token
+from tests.integration.integration_test_case import IntegrationTestCase
+
+
+class TestQuestionnairePreviousLink(IntegrationTestCase):
+
+    def test_previous_link_doesnt_appear_on_introduction(self):
+        # Given
+        token = create_token('final_confirmation', 'test')
+        resp = self.client.get('/session?token=' + token.decode(), follow_redirects=False)
+
+        # When we open the introduction
+        content = resp.get_data(True)
+
+        # Then previous link does not appear
+        self.assertNotRegexpMatches(content, 'Previous')
+
+    def test_previous_link_appears_on_all_non_start_end_pages(self):
+        base_url = '/questionnaire/test/final_confirmation/789/'
+
+        # Given
+        token = create_token('final_confirmation', 'test')
+        self.client.get('/session?token=' + token.decode(), follow_redirects=False)
+
+        # When
+        # We proceed through the questionnaire
+        post_data = {
```

Comment:
Yeah, the form is posting back - so I'm not sure why we'd change the integration test to do something else.